website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+from openerp import http
+from openerp.http import request
+from openerp.osv import orm
+import json
+
+
+class Chd_website(http.Controller):
+
+    @http.route('/chd_init/', auth='public', website=True)
+    def start(self, selected_id=False, type=False):
+        partner = self.get_current_partner()
+        if partner is False:
```

Comment:
this checks for partner being exactly False, but you probably also want to have that for an empty recordset, so better only check for falsyness: `if not partner:`